core/authentication_api.php

```diff
@@ -764,30 +790,35 @@ function auth_does_password_match( $p_user_id, $p_test_password ) {
  * @access public
  */
 function auth_process_plain_password( $p_password, $p_salt = null, $p_method = null ) {
-	$t_login_method = config_get( 'login_method' );
-	if( $p_method !== null ) {
-		$t_login_method = $p_method;
-	}
+	$t_login_method = $p_method !== null ? $p_method : config_get( 'login_method' );
 
 	switch( $t_login_method ) {
 		case CRYPT:
-
 			# a null salt is the same as no salt, which causes a salt to be generated
 			# otherwise, use the salt given
 			$t_processed_password = crypt( $p_password, $p_salt );
 			break;
 		case MD5:
 			$t_processed_password = md5( $p_password );
 			break;
+		case HASH_BCRYPT:
```

Comment:
How about either also adding `LOGIN_METHOD_HASH_MD5`, `LOGIN_METHOD_HASH_CRYPT`, `LOGIN_METHOD_HASH_CRYPT_FULL_SALT` and using them in the code, though keeping `MD5` and `CRYPT` and `CRYPT_FULL_SALT` for backward compatibility?  Or we can just rename this constant to BCRYPT.